setup-provider.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# QA has a new process to test Terraform using rcs. We will not publish rcs
+# onto the Terraform registry because Hashicorp could potentially block our
+# testing by being slow to publish.
+
+# Instead, we will test using a downloaded binary from the rc. This script
+# sets up the correct binary using a defined <provider> <version> to test
+# updates locally.
+
+# Example
+# ./setup-provider.sh rancher2 v3.0.0-rc1
+
+# Validate user input
+
+if [ -z "$1" ] || [ -z "$2" ]; then
+    echo -e "Please specify a Terraform provider and version tag."
+    echo -e "./setup-provider.sh <provider> <version>"
+    exit 1
+fi
+
+# Set global vars
+
+PROVIDER=$1
+VERSION=$2
+VERSION_TAG=$(echo $2 | cut -c 2-)
+BASE=~/.terraform.d/plugins/terraform.example.com/local/${PROVIDER}/${VERSION_TAG}
+
+# Install gzip
+
+brew install gzip
```

Comment:
Bit of a nit. Can we add a conditional to check the if it's already installed? If not, then install `gzip`. I am assuming QA will use a VM with this script (if not, different story lol), but in the scenario they don't, this may be an issue. Some members use Windows as their local, so this installation and the case statement below is an issue

I don't see it being an issue, just calling it out.